spec/factorial_spec.rb

```diff
@@ -0,0 +1,19 @@
+require_relative '../solver'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  it 'should test when given number is negative' do
+    expect { @solver.factorial(-5) }.to raise_error(RuntimeError)
+  end
```

Comment:
- [ ] Please make sure if a negative integer is given it should raise an exception. It is part of the[ project requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md)